src/metabase/api/setup.clj

```diff
@@ -38,21 +38,33 @@
   (su/with-api-error-message (s/constrained su/NonBlankString setup/token-match?)
     "Token does not match the setup token."))
 
+(def ^:dynamic ^:private *disallow-api-setup-after-first-user-is-created*
+  "We must not allow users to setup multiple super users once the first user is created. But tests still need to be able
+  to do that. This var is redef'd to false by certain tests to allow that."
+  true)
+
 (defn- setup-create-user! [{:keys [email first-name last-name password]}]
+  (when (and *disallow-api-setup-after-first-user-is-created* (setup/has-user-setup))
+    ;; many tests use /api/setup to setup multiple users, so *first-user-only* is redefined by them
+    (throw (ex-info
+            (tru "The /api/setup route can only be used to create the first user, however a user currently exists.")
+            {:status-code 403
```

Comment:
I'm wondering if we should just return the 403 here and not the other information.

Reply:
I noticed some other places ex-info used return information, but perhaps the `has-users-setup` value is too internal to be useful here.